### DISCUSSION TITLE
AOSP 12.0 v401

### DISCUSSION THREAD

**Discussion by phhusson** [phhusson/treble_experimentations#2173] [General]
- @ponces and me spent sleep-less nights on it, but here we are, we now have OTA \o/ (only on dynamic partition devices)
- February Security Patch Level
- Workaround an issue on  some Samsung devices where touchscreen would be dead on resume
- Automatically enable debug.sf.latch_unsignaled=1 on devices reporting that in vendor prop -- Should improve graphical perf on some devices
- Fix in-call BT voice on Moto One Vision
- Fix boot on Snapdragon 8 gen 1/Android 12 vendors/audio hal 7.0
- Enable WV L1 in Netflix on Mi 11 Lite 5G (Thanks @pessip)
- Add an option in treble settings to restart SystemUI (currently requires root), useful when you change resolution (Thanks @AndyCGYan )
- Add support for Huawei IMS (Thanks @iceows)
- Allow volume buttons to control volume of remote devices (thanks @ponces )

<hr /><em>This discussion was created from the release <a href='https://github.com/phhusson/treble_experimentations/releases/tag/v401'>AOSP 12.0 v401</a>.</em>

**Comment by 09kkthxbb** [comment]
Thanks for the continuous development and effort! Salute to you guys 😄

**Comment by NofanTasi** [comment]
Is it recommended to factory reset when going 11 -> 12 ? Or is dirty flash (wipe dalvik and cache) an option?
Thanks for effort. My Xiaomi MiA1 runs 11 a/b Treble and came from 7 (and was not even a/b) :-)

**Reply by devrieshh** [reply]
no,just flash the system

**Reply by devrieshh** [reply]
just do a factory reset if it wont boot

**Reply by NofanTasi** [reply]
@devrieshh : thanks for attention ... FWIW: even with factory reset Phh squeak ab gapps 401 A12 is stuck in Christmas animation on MiA1. I reverted back to roar ab gapps 313 A11. On Telegram someone says a fix may be needed in **vendor**. So it is not a Phh issue.

**Reply by devrieshh** [reply]
cant you flash Vendor.img. and  boot.img from stock ROM over v401

**Reply by NofanTasi** [reply]
Stock ROM never had a stock vendor as it was never stock treble. A repartition via TWRP was made possible by XDA devs to make the MiA1 treble, and, some vendor and boot images were made by XDA devs to work with a GSI system image. It worked well for A9, A10 and A11, but, not for A12. In any case: it is not a Phh system issue. I was told it is a (non stock) vendor issue.

**Reply by devrieshh** [reply]
try flashing vendor.img and than boot.img     from your stock (original)rom  after flashing    AOSP 12.0 v402

**Reply by NofanTasi** [reply]
@devrieshh : thanks for support. As you know: non-treble aware (stock) kernels do not work. But we do have (several) treble aware kernels. I will certainly try AOSP 12.0 v402 with non-stock sooti vendor.img and different non-stock treble aware kernels. Have a nice day.

**Comment by ghost** [comment]
!!OTA UPDATES!!
😱🙀🤯
ALL THIS DEV ARE GOD 
Thanks 😍 
Il definitely ganna try this on OP 8 PRO

**Comment by tnolte1** [comment]
Thanks for the OTA updates! Really great! That's really a reason to buy a new device with dynamic partition support ;)

The christmas boot animation is  a little outdated since it's Feb. now, though I liked it a lot ;)

**Comment by lostmsu** [comment]
How do I know if the device is a "dynamic partitions" device?

**Reply by phhusson** [reply]
Was your device launched with Android 10?
If yes, then it's a dynamic partitions devices
If not, then it's not.

**Reply by lostmsu** [reply]
@phhusson so this is not the same as `A/B` vs `A-only`?

Asking because some devices have "A/B when updated to Android 10" comment in Wiki.

**Reply by tnolte1** [reply]
This would be really cool if it was available also for Android 10 devices, as this is what i have.
My initial understanding was the same as lostmsu that it is available only for devices with seamless updates, i.e. Android 11.

**Reply by phhusson** [reply]
@lostmsu no it's not the same. The device must be **launched** with Android 10

@tnolte1 It is available to Android 10 devices. This doesn't require the device to support seamless updates, even though this OTA mechanism is seamless.

**Comment by Taogezuibang** [comment]
Hello, thank you very much for your efforts.
I come from China. The equipment I use is redmi 10x pro, and the certified model is m2004j7bc
Some problems encountered during use:
1. After you turn on volte, you can make a call, but you can't answer the call because there is no response
2. The screen fingerprint can only be entered with the highest brightness, which is slow and insensitive
3. After the eye protection mode is turned on, the screen will flash continuously

I hope it can be solved later. Thank you
